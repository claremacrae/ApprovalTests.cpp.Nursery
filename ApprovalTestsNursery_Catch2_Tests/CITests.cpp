#include "Catch.hpp"
#include "ApprovalTests/Approvals.h"
#include "ApprovalTests/namers/ApprovalNamer.h"

#include <cstdio>

TEST_CASE("Fail with ApprovalMissingException")
{
    auto namer = Approvals::getDefaultNamer();

    // Delete auto-created Approved File, in case left over from previous run
    auto approved = namer->getApprovedFile(".txt");
    remove(approved.c_str());
    
    try
    {
        Approvals::verify("Do not approve me - approved file must not exist");
        // In Travis and Appveyor, neither finds a diffing tool
        // So neither blocks
        // However, they don't write out what the differences are.
    }
    catch(const ApprovalMissingException&)
    {
        // We intend to arrive here

        // Re-delete auto-created Approved File, so it doesn't get checked
        // in to git
        remove(approved.c_str());
    }
}

TEST_CASE("Fail with ApprovalMismatchException")
{
    try
    {
        Approvals::verify("I will fail - do not approve me");
    }
    catch(const ApprovalMismatchException&)
    {
        // We intend to arrive here
    }
}
