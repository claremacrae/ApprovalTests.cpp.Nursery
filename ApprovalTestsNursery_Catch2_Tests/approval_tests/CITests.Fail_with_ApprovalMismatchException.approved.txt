I have different content to what is being approved in the source code
